netcore/System.Private.CoreLib/src/System/Threading/LowLevelLifoSemaphore.Unix.Mono.cs

```diff
@@ -60,9 +64,9 @@ private bool WaitCore (int timeoutMs)
 			}
 
 			try {
-				Monitor.try_enter_with_atomic_var (wait_entry.condition, Timeout.Infinite, false, ref waitEntryLocked);
+				Monitor.try_enter_with_atomic_var (ref wait_entry.condition, Timeout.Infinite, allowInterruption, ref waitEntryLocked);
```

Comment:
this is pretty fragile - `wait_entry` happens to be a struct, so it's heap allocated, so `ref wait_entry.condition` is a pointer to the stack (and so whatever `condition` points to is pinned), but if `wait_entry` were a class, it wouldn't be.